include/zephyr/bluetooth/audio/micp.h

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2020-2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_MICP_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_MICP_H_
+
+/**
+ * @brief Microphone Input Control Profile (MICP)
+ *
+ * @defgroup bt_gatt_micp Microphone Input Control Profile (MICP)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * [Experimental] Users should note that the APIs can change
+ * as a part of ongoing development.
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/bluetooth/audio/aics.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(CONFIG_BT_MICP_MIC_DEV)
+#define BT_MICP_MIC_DEV_AICS_CNT CONFIG_BT_MICP_MIC_DEV_AICS_INSTANCE_COUNT
+#else
+#define BT_MICP_MIC_DEV_AICS_CNT 0
+#endif /* CONFIG_BT_MICP_MIC_DEV */
+
+/** Application error codes */
+#define BT_MICP_ERR_MUTE_DISABLED                  0x80
+#define BT_MICP_ERR_VAL_OUT_OF_RANGE               0x81
+
+/** Microphone Input Control Profile mute states */
+#define BT_MICP_MUTE_UNMUTED                       0x00
+#define BT_MICP_MUTE_MUTED                         0x01
+#define BT_MICP_MUTE_DISABLED                      0x02
+
+/** @brief Opaque Microphone Controller instance. */
+struct bt_micp_mic_ctlr;
+
+/** @brief Register parameters structure for Microphone Input Control Service */
+struct bt_micp_mic_dev_register_param {
+#if defined(CONFIG_BT_MICP_MIC_DEV_AICS)
+	/** Register parameter structure for Audio Input Control Services */
+	struct bt_aics_register_param aics_param[BT_MICP_MIC_DEV_AICS_CNT];
+#endif /* CONFIG_BT_MICP_MIC_DEV_AICS */
+
+	/** Microphone Input Control Profile callback structure. */
+	struct bt_micp_mic_dev_cb *cb;
+};
+
+/**
+ * @brief Microphone Input Control Profile included services
+ *
+ * Used for to represent the Microphone Input Control Profile included service
+ * instances, for either a Microphone Controller or a Microphone Device.
+ * The instance pointers either represent local service instances,
+ * or remote service instances.
+ */
+struct bt_micp_included {
+	/** Number of Audio Input Control Service instances */
+	uint8_t aics_cnt;
+	/** Array of pointers to Audio Input Control Service instances */
+	struct bt_aics **aics;
+};
+
+/**
+ * @brief Initialize the Microphone Input Control Profile Microphone Device
+ *
+ * This will enable the Microphone Input Control Service instance and make it
+ * discoverable by Microphone Controllers.
+ *
+ * @param param Pointer to an initialization structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_micp_mic_dev_register(struct bt_micp_mic_dev_register_param *param);
```

Comment:
> My default is to keep naming as is here.

Same

IMO `micp_dev` and `micp_ctlr` is "as bad" as just using `bap_source` and `bap_client` (especially the former, since "device" is very generic).